### DISCUSSION TITLE
Question about Isotropic Power Hardening stress update

### DISCUSSION THREAD

**Discussion by Bala-1005** [idaholab/moose#17580] [Q&A Modules: Solid mechanics] [answered]
Hello everyone,

I have a question about how IsotropicPowerHardeningStressUpdate material is setup. As far as I can see, this particular class is used to calculate the yield stress based on the strength coefficient and strain hardening exponent. My question is how is the inelastic strain calculated while using this class? Is it directly inherited from IsotropicPlasticityStressUpdate?

Thanks,
Bala

**Comment by hugary1995** [comment, marked as answer]
The `IsotropicPowerHardeningStressUpdate` inherits from `IsotropicPlasticityStressUpdate`. The inelastic strain is calculated using the return mapping algorithm.
